### PR TITLE
Csi 1156 improve database operation performace

### PR DESCRIPTION
**Pull request by HarishRocks**
Previous Sync time in total for GMT wallet with all coins: 150s
In this fix: 58s
more than 60% reduction in time

### CODE REVIEW

**Comment by HarishRocks** [comment]
https://github.com/Cypherock/cysync-desktop/pull/170/files#diff-02a6a8170f283b42256892114a921b915885392a88a55e7a3c58a00881dbc5d2L377

@irshadCypherock I see we are checking for existing txns while inserting/updating txns. Can you forsee if batching the inserts like I had done would cause any problem?

**Comment by irshadCypherock** [comment]
Task linked: [CSI-1156 Improve Database Operation Performace](https://app.clickup.com/t/37308523/CSI-1156)

File: cysync/src/store/database/helper/transaction/insert.ts [hunk 1]
```diff
@@ -528,7 +528,7 @@ export const insertFromBlockbookTxn = async (transaction: {
         outputs: []
       };
 
-      await transactionDb.insert(feeTxn);
+      return feeTxn;
```
Comment (irshadCypherock) [inline cysync/src/store/database/helper/transaction/insert.ts:531]: In this case both the fees and the txn below should be added, which will not happen after this change.

File: cysync/src/store/provider/syncProvider/executors/history.ts [hunk 2]
```diff
@@ -217,6 +220,8 @@ export const processResponses = async (
         };
       }
     }
+
+    await transactionDb.insertMany(transactionDbList);
```
Comment (irshadCypherock) [inline cysync/src/store/provider/syncProvider/executors/history.ts:224]: Catch errors for this and log

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: cysync/src/store/database/helper/transaction/insert.ts (hunk 1 above)
Comment (HarishRocks) [reply cysync/src/store/database/helper/transaction/insert.ts:531]: Nice catch!

File: cysync/src/store/provider/syncProvider/executors/history.ts (hunk 2 above)
Comment (HarishRocks) [reply cysync/src/store/provider/syncProvider/executors/history.ts:224]: yeap got it

**Review by Suraj-Tiwari — APPROVED** [review]

**Pull request merged**